File: ThefieldofdreamsClient/src/app/models/mission.ts
export type MissionType =
  | 'PredictMatch'
  | 'ScanQRCode'
  | 'ShareReferral'
  | 'CompleteTourismRoute'
  | 'InviteFriend'
  | 'DailyLogin'
  | 'ReachLeaderboardRank'
  | 'MakePredictionStreak';

export type MissionDifficulty = 'Easy' | 'Medium' | 'Hard' | 'Expert';

export interface Mission {
  id: string;
  title: string;
  description?: string;
  type: MissionType;
  requiredProgress: number;
  pointsReward: number;
  xpValue?: number;
  difficulty: MissionDifficulty;
  startDate?: string;
  endDate?: string;
  isDaily: boolean;
  requiredUserLevel?: number;
}

export interface UserMission {
  id: string;
  userId: string;
  missionId: string;
  currentProgress: number;
  isCompleted: boolean;
  completedAt?: string;
  pointsAwarded: number;
}

export interface CreateMissionRequest {
  title: string;
  description?: string;
  type: MissionType;
  requiredProgress: number;
  pointsReward: number;
  xpValue?: number;
  difficulty: MissionDifficulty;
  startDate?: string;
  endDate?: string;
  isDaily: boolean;
  requiredUserLevel?: number;
}

export interface UpdateMissionRequest extends CreateMissionRequest {}

export interface UpdateUserMissionProgressRequest {
  currentProgress: number;
  isCompleted: boolean;
  pointsAwarded: number;
}

File: Thefieldofdreams.Domain/Entities/Mission.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Thefieldofdreams.Domain.Entities
{
    

    public class Mission : BaseEntity
    {
        public required string Title { get; set; }
        public string? Description { get; set; }
        public MissionType Type { get; set; }
        public int RequiredProgress { get; set; } = 1;
        public int PointsReward { get; set; }
        public int? XPValue { get; set; }
        public MissionDifficulty Difficulty { get; set; } = MissionDifficulty.Medium;
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public bool IsDaily { get; set; } = false;
        public int? RequiredUserLevel { get; set; }

        public virtual ICollection<UserMission> UserMissions { get; set; } = new List<UserMission>();
    }

    public enum MissionType
    {
        PredictMatch,
        ScanQRCode,
        ShareReferral,
        CompleteTourismRoute,
        InviteFriend,
        DailyLogin,
        ReachLeaderboardRank,
        MakePredictionStreak
    }

    public enum MissionDifficulty
    {
        Easy,
        Medium,
        Hard,
        Expert
    }
}
